internal/pkg/agent/application/monitoring/stats.go

```diff
@@ -12,6 +12,35 @@ import (
 	"github.com/elastic/elastic-agent-libs/monitoring"
 )
 
+const formValueKey = "failon"
+
+type LivenessFailConfig struct {
+	Degraded bool `yaml:"degraded" config:"degraded"`
+	Failed   bool `yaml:"failed" config:"failed"`
+}
+
+// process the form values we get via HTTP
+func handleFormValues(req *http.Request) (LivenessFailConfig, error) {
+	err := req.ParseForm()
+	if err != nil {
+		return LivenessFailConfig{}, fmt.Errorf("Error parsing form: %w", err)
+	}
+
+	defaultUserCfg := LivenessFailConfig{Degraded: false, Failed: true}
+
+	userConfig := req.Form.Get(formValueKey)
+	if userConfig == "failed" {
+		// same as default
+		return defaultUserCfg, nil
+	} else if userConfig == "degraded" {
+		return LivenessFailConfig{Failed: true, Degraded: true}, nil
+	} else if userConfig == "" {
+		return defaultUserCfg, nil
+	} else {
+		return defaultUserCfg, fmt.Errorf("got unexpected value for `%s` attribute: %s", formValueKey, userConfig)
+	}
```

Comment:
a switch statement would probably be easier to read, especially since you have "" and "failed" doing the same thing.

What if the user doesn't want a "failed" component to mean that elastic-agent is dead?  I don't think you can get that behavior with this code.